src/BloomBrowserUI/bookEdit/BookPagesThumbnailList/BookPagesThumbnailList.htm

```diff
@@ -31,12 +31,36 @@
             jQuery('.gridItem').click(function () {
                 fireCSharpEvent("gridClick", $(this).attr('id'));
             });
+
+            setTimeout(loadNextThumbnail, 100);
         });
         function fireCSharpEvent(eventName, eventData) {
 
             var event = new MessageEvent(eventName, { 'view': window, 'bubbles': true, 'cancelable': true, 'data': eventData });
             document.dispatchEvent(event);
         }
+
+        function loadNextThumbnail() {
+
+            // The "thumb-src" attribute is added to the img tags on the server while the page is being built. The value
+            // of the "src" attribute is copied into it and then the "src" attribute is set to an empty string so the
+            // images can be loaded here in a controlled manner so as not to overwhelm system memory.
+            var nextImg = jQuery('body').find('img[thumb-src]').first();
+
+            // stop processing if there are no more images
+            if ((!nextImg) || (nextImg.length === 0)) return;
+
+            var img = nextImg[0];
+
+            // adding this to the query string tells the server to generate a thumbnail from the image file
+            var src = img.getAttribute('thumb-src') + '?thumbnail=1';
+            img.removeAttribute('thumb-src');
+            img.src = src;
+
+            // This delay is needed because the processing of larger images was causing out-of-memory errors if several
+            // images were requested before the server had finished processing previous images requests.
+            setTimeout(loadNextThumbnail, 100);
```

Comment:
Maybe explain why we need this as WELL as the code that limits us to handling #cores requests at once.

I believe part of the idea is that we don't want the browser to time-out and display a missing-image thing if we can't produce thumbnails fast enough, but that isn't mentioned here.